### PR TITLE
Sign up use gem devise

### PR DESCRIPTION
**Pull request by vuducdat**

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class RegistrationsController < Devise::RegistrationsController
+
+  private
+
```
Comment (phuongbk) [inline app/controllers/registrations_controller.rb]: Không cách!

File: app/views/devise/registrations/_signup.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, class: "form-control" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
```
Comment (leminhtuan2015) [inline app/views/devise/registrations/_signup.html.erb]: I18n pls

File: app/views/devise/registrations/_signup.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= devise_error_messages! %>
+
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.email_field :name, autofocus: true, class: "form-control" %>
```
Comment (phuongbk) [inline app/views/devise/registrations/_signup.html.erb]: email_fied for name ???

File: app/views/devise/registrations/_signup.html.erb (hunk 2 above)
Comment (phuongbk) [inline app/views/devise/registrations/_signup.html.erb]: I18n?

File: app/views/devise/registrations/_signup.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
```
Comment (leminhtuan2015) [inline app/views/devise/registrations/_signup.html.erb]: Do you need ( ) ?

File: app/views/devise/registrations/_signup.html.erb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  <%= devise_error_messages! %>
```
Comment (phuongbk) [inline app/views/devise/registrations/_signup.html.erb]: rm ()

File: app/views/devise/registrations/_signup.html.erb (hunk 4 above)
Comment (phuongbk) [inline app/views/devise/registrations/_signup.html.erb]: rm ()

File: app/views/devise/registrations/_signup.html.erb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.email_field :name, autofocus: true, class: "form-control" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, class: "form-control" %>
```
Comment (phuongbk) [inline app/views/devise/registrations/_signup.html.erb]: duplicate autofocus

File: app/views/devise/registrations/_signup.html.erb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, class: "form-control" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> <%= t "user.characters_minimum" %>)</em>
```
Comment (dieunb) [inline app/views/devise/registrations/_signup.html.erb]: indent

**Pull request merged**